Classes/Service/FileGenerator.php

```diff
@@ -240,7 +238,11 @@ public function build(Extension $extension)
             $this->generateDocumentationFiles();
         }
 
-        $this->generateEmptyGitRepository();
+        $extensionSettings = $extension->getSettings();
+        if ($extensionSettings['overwriteSettings']['.git'] ?? false) {
```

Comment:
the null coalesce operator tests for null only. 
So this condition will be true for any value like false, overwrite etc. 
I would prefer a test like === 'skip'
but I am open for other opinions... ;-)

Reply:
The correct condition must be:
`if (!isset($extensionSettings['overwriteSettings']['.git']) || $extensionSettings['overwriteSettings']['.git'] !== 'skip') {`